folly/SocketPortability.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2015 Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+// While MSVC may not have Unix sockets, Cygwin does.
+#ifdef _MSC_VER
+# define HAVE_UNIX_SOCKETS 0
+#else
+# define HAVE_UNIX_SOCKETS 1
+#endif
+
+#ifdef _MSC_VER
+
+#ifndef __STDC__
```

Comment:
I assume there's some way to disable this one? I need that symbol to be named that way.